src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt

```diff
@@ -89,6 +96,31 @@ class CHIPToolActivity :
         .commit()
   }
 
+  private fun onNfcIntent(intent: Intent?) {
+    // Require 1 NDEF message containing 1 NDEF record
+    val messages = intent?.getParcelableArrayExtra(NfcAdapter.EXTRA_NDEF_MESSAGES)
+    if (messages?.size != 1) return;
+
+    val records = (messages[0] as NdefMessage).records
+    if (records.size != 1) return;
+
+    // Require NDEF URI record starting with "ch:"
+    val uri = records[0].toUri()
```

Comment:
What happens with spaces, especially when they appear at the end of the string?

Reply:
Seems to be working correctly, but I agree we should bring that up during spec review (which I saw you had already done).